### PR TITLE
Fp

### PR DESCRIPTION
**Pull request by Egorles**

### CODE REVIEW

File: test/exercise/fp/solution.rb [hunk 1]
```diff
@@ -4,12 +4,28 @@ class << self
...
-      def rating(_array)
-        0
+      def rating(array)
+        array_raitings_with_nil = array.map do |film|
+          if film['country'].to_s.include?(',') && film['rating_kinopoisk'].to_f.positive?
+            film['rating_kinopoisk'].to_f
+          end
+        end
+
+        array_raitings = array_raitings_with_nil.select { |num| !num.nil? }
```
Comment (artempartos) [inline test/exercise/fp/solution.rb]: select !nil => compact

File: test/exercise/fp/solution.rb [hunk 2]
```diff
@@ -4,12 +4,28 @@ class << self
...
+
+        sum_rate = array_raitings.reduce(:+)
```
Comment (artempartos) [inline test/exercise/fp/solution.rb]: array_raitings -> array_ratings )

File: test/exercise/fp/solution.rb [hunk 3]
```diff
@@ -4,12 +4,28 @@ class << self
       # Обратиться к параметрам фильма можно так:
       # film["name"], film["rating_kinopoisk"], film["rating_imdb"],
       # film["genres"], film["year"], film["access_level"], film["country"]
-      def rating(_array)
-        0
+      def rating(array)
+        array_raitings_with_nil = array.map do |film|
+          if film['country'].to_s.include?(',') && film['rating_kinopoisk'].to_f.positive?
```
Comment (artempartos) [inline test/exercise/fp/solution.rb]: lets select them map

File: test/exercise/fp/solution.rb [hunk 4]
```diff
@@ -4,12 +4,28 @@ class << self
...
+
+        sum_rate = array_raitings.reduce(:+)
+
+        middle_rate = sum_rate / array_raitings.length
       end
 
-      def chars_count(_films, _threshold)
-        0
+      def chars_count(films, threshold)
+        array_films_names_with_nil = films.map { |film| film['name'].to_s if film['rating_kinopoisk'].to_f >= threshold }
```
Comment (artempartos) [inline test/exercise/fp/solution.rb]: lets select them map

**Review by artempartos — CHANGES_REQUESTED** [review]

File: test/exercise/fp/solution.rb (hunk 2 above)
Comment (Egorles) [reply test/exercise/fp/solution.rb]: @artempartos Yeeep...

File: test/exercise/fp/solution.rb (hunk 3 above)
Comment (Egorles) [reply test/exercise/fp/solution.rb]: @artempartos "then"?) 
> array_raitings -> array_ratings )

  1:1  ;D

File: test/exercise/fp/solution.rb (hunk 4 above)
Comment (Egorles) [reply test/exercise/fp/solution.rb]: @artempartos
Sorry, din't get it... Should I split up array.map into array.select and then array.map  ?

**Review by artempartos — APPROVED** [review]

File: test/exercise/fp/solution.rb (hunk 3 above)
Comment (artempartos) [reply test/exercise/fp/solution.rb]: )

**Pull request merged**